src/Actions/JoinLines.ts

```diff
@@ -63,6 +63,13 @@ export class ActionJoinLines {
             ranges.forEach(range => {
                 editBuilder.replace(range, ' ');
             });
+
+            // move to the space between the last two joined lines
+            linesToJoin.sort();
+            const lastLine = linesToJoin[linesToJoin.length - 1];
+            const char = activeTextEditor.document.lineAt(lastLine).text.length;
+            const position = new Position(lastLine, char)
+            activeTextEditor.selection = new Selection(position, position);
```

Comment:
I've thought about multicursor positions. It's rather difficult. I think it would be good to define the desired behaviour.

I checked out Vim's multiselect, and I as far as I can see it removes all cursors except for the last one and positions the single cursor where the last 'join line' occurred. The code I committed does the same thing - you can see it in the last Visual J test: `[foo\nb]ar\nabc\n[123\n4]56\n789`.

I also looked at VSCodeVim's behaviour, and well, that doesn't behave very well with multiple cursors and J ;-)

IMHO, I believe the current behaviour is 'good enough', but I'm also willing to work with you to define the best solution.

Reply:
I just tried in Vim (with https://github.com/terryma/vim-multiple-cursors), cursors are preserved at different positions.
Would be great we do the same.
![2018-06-06 14 04 58](https://user-images.githubusercontent.com/2175401/41018011-b1d0b83e-6992-11e8-97fd-9e1692d3feb3.gif)

Reply:
One solution will be:

```ts
const targetPositions: Position[] = [];
const linesToJoin: number[] = [];

activeTextEditor.selections.forEach(selection => {
    if (selection.isSingleLine) {
        const line = activeTextEditor.document.lineAt(selection.active.line);
        targetPositions.push(new Position(line.lineNumber, line.text.length));

        linesToJoin.push(line.lineNumber);
    }
    else {
        const line = activeTextEditor.document.lineAt(selection.end.line - 1);
        targetPositions.push(new Position(line.lineNumber, line.text.length));

        for (let lineNumber = selection.start.line; lineNumber <= line.lineNumber; lineNumber++) {
            linesToJoin.push(lineNumber);
        }
    }
});

activeTextEditor.selections = targetPositions.map((position) => new Selection(position, position));

// ...
```

Reply:
So I was looking at a different Vim multiselect plugin: https://www.vim.org/scripts/script.php?script_id=953

Thanks for your solution. Your code works perfectly, of course ;-)